### PR TITLE
Add schema for gene families endpoint

### PR DESCRIPTION
**Pull request by gnilekaw**
As part of the new [Categories page](https://www.artsy.net/categories), we'd like to fetch gene families from the new endpoint. This PR adds a new schema to that end. 

As this is my first non-trivial PR for Metaphysics/GraphQL, there may be some learning opportunities here. So please point out ways in which this  PR could be improved if you see them. 

CC @oxaudo @anandaroop

### CODE REVIEW

File: schema/gene_families.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import gravity from "../lib/loaders/gravity"
+import GeneFamily from "./gene_family"
+import { GraphQLList } from "graphql"
+
+const GeneFamilies = {
+  type: new GraphQLList(GeneFamily.type),
+  description: "A list of Gene Families",
+  resolve: () => gravity("gene_families"),
```
Comment (alloy) [inline schema/gene_families.js]: Can you replace the direct use of the `gravity` loader with defining a specific loader [here](https://github.com/artsy/metaphysics/blob/master/lib/loaders/per_type.js)? Your resolve function would then be defined like so:

```js
resolve:  (_source, _args, _request, { rootValue: { geneFamiliesLoader } }) => geneFamiliesLoader(),
```

File: schema/gene_family.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import type { GraphQLFieldConfig } from "graphql"
+import { GraphQLObjectType, GraphQLString, GraphQLList } from "graphql"
+import Gene from "./gene"
+
+const GeneFamilyType = new GraphQLObjectType({
+  name: "GeneFamily",
+  fields: {
+    id: {
+      type: GraphQLString,
+    },
```
Comment (alloy) [inline schema/gene_family.js]: Can this field be `null`? If not, define it as such with `new GraphQLNonNull(GraphQLString)`.

File: schema/gene_family.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    name: {
+      type: GraphQLString,
+    },
```
Comment (alloy) [inline schema/gene_family.js]: Same.

File: schema/gene_family.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      type: new GraphQLList(Gene.type),
+    },
+  },
+})
+
+const GeneFamily: GraphQLFieldConfig<GeneFamilyType, *> = {
+  type: GeneFamilyType,
+  resolve: () => {
+    return root => root
+  },
```
Comment (alloy) [inline schema/gene_family.js]: Why do you return an (identity) function here?

File: test/schema/gene/gene_families.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    const query = `
+      {
+        gene_families {
+          id
+          name
+        }
+      }
+    `
+
+    return runQuery(query).then(data => {
```
Comment (alloy) [inline test/schema/gene/gene_families.js]: Once you’ve replaced the use of the `gravity` loader with a per-type loader you can get rid of all the rewiring in this test case. Instead you you use DI by passing in a stub function to [the GraphQL query execution](https://github.com/artsy/metaphysics/blob/2d9de40d40a537a17447d7094589f8c9065e17ba/test/utils.js#L17) that acts as your specific loader:

```js
const geneFamiliesLoader = () => Promise.resolve(api_data)
runQuery(query, { geneFamiliesLoader })
```

File: schema/gene_family.js (hunk 4 above)
Comment (gnilekaw) [inline schema/gene_family.js]: Good question. I mistakenly thought `resolve` was required. But it can be cut completely.

Comment (alloy) [inline schema/gene_family.js]: Orly?

I once [returned an empty object](https://github.com/artsy/metaphysics/blob/2d9de40d40a537a17447d7094589f8c9065e17ba/schema/home/index.js#L23-L26), but if it’s really not needed at all then can you remove it there too?

Comment (gnilekaw) [inline schema/gene_family.js]: It actually is needed right in the `home/index` instance!

**Comment by gnilekaw** [comment]
@alloy Great comments. Thank you. I was able to cut down the number of lines even more 🔪

File: schema/gene_family.js (hunk 4 above)
Comment (alloy) [inline schema/gene_family.js]: How quirky!

**Comment by alloy** [comment]
Awesome, looks perfect 👌✨

**Pull request merged**